### ISSUE TITLE
How can model update weights when trainable_model is trained (DQN)

### ISSUE THREAD

**Issue by MaxHoefl** [keras-rl/keras-rl#315]
I am quite confused how in `rl/agents/dqn.py`, the target model `self.model` updates its weights when it is `self.trainable_model` on which `fit_on_batch` is called (e.g. line 325 in `dqn.py`). 

The only explanation that I can think of is that the loss tensor (`loss_out`) for `self.trainable_model`  is computed using the predictions from `self.model` (`y_pred` tensor) and hence the computational graph sends the gradients through `y_pred` into the weights of `self.model`. But if this is the case then I am quite astonished.

Could someone explain please?

Thanks for the help
